structs.go

```diff
@@ -959,7 +959,7 @@ type Identify struct {
 	Shard              *[2]int             `json:"shard,omitempty"`
 	Presence           GatewayStatusUpdate `json:"presence,omitempty"`
 	GuildSubscriptions bool                `json:"guild_subscriptions"`
-	Intents            int                 `json:"intents"`
+	Intents            *Intent             `json:"intents,omitempty"`
```

Comment:
I think leaving the field as an `int` as fine. `0` falls under the `omitempty` clause for ints; introducing it as a pointer makes doing initialization a bit more difficult.

Reply:
There are a few events that are still sent for a `0` intent, and `omitempty` on a pointer still allows a 0 to be sent over the wire.

Instead, add a utility function that heap-allocates the integer:

```
func MakeIntent(intents Intent) *Intent {
       return &intents
}
```

Reply:
This seems fine for now, I like the idea of the Helper method.

Reply:
Ye, the 0 is valid intent thats why I used pointer. I added helper method to utils.